### PR TITLE
Update balena-auth from 4.0.2 to 4.1.0

### PR DESCRIPTION
**Pull request by josecoelho**
Update balena-errors from 4.6.0 to 4.7.1

- [x] Do not merge until required PRs are merged and the versions updated.

Required by balena-io/balena-cli#2050

Depends-on: https://github.com/balena-io-modules/balena-auth/pull/26
Depends-on: https://github.com/balena-io-modules/balena-request/pull/152
Change-type: minor

### CODE REVIEW

**Comment by josecoelho** [comment]
@pdcastro It's failing with `throw new Error('Missing environment credentials');` which might be an ENV issue. The tests are passing locally 🤔

**Comment by josecoelho** [comment]
I noticed that the `build` folder is not on `.gitignore` as well, shall we add it?

**Comment by pdcastro** [comment]
@thgreasi is the expert in the SDK territory... Perhaps he can answer these questions.

**Comment by thgreasi** [comment]
> It's failing with `throw new Error('Missing environment credentials');` which might be an ENV issue. The tests are passing locally

@josecoelho it's failing b/c the branch is in your personal clone of the repo `josecoelho:update-balena-auth`. Push a branch to this repo and open a new PR plz.

The `build` folder is not in the `.gitignore` b/c we are not using it :) We are emitting `es2015` & `es2018` builds in the respective folders, which are in the `.gitignore`.

**Comment by pdcastro** [comment]
> Push a branch to this repo and open a new PR plz.

Do external contributors have permissions for this? José is not in the balena team.

**Comment by thgreasi** [comment]
Oh I didn't know that @pdcastro, thanks for the heads up!
Huge thanks for pushing this that far @josecoelho.
Let me open a same-repo PR to confirm that the tests pass.

**Comment by thgreasi** [comment]
Tests started at
See: https://ci.appveyor.com/project/resin-io/balena-sdk/builds/36252496

**Comment by thgreasi** [comment]
@josecoelho the tests pass :tada: 
Feel free to mark the PRs as ready for review and update the dependencies in the follow-ups!

File: package.json [hunk 1]
```diff
@@ -99,12 +99,12 @@
...
     "abortcontroller-polyfill": "^1.5.0",
-    "balena-auth": "^4.0.2",
-    "balena-errors": "^4.6.0",
+    "balena-auth": "^4.1.0",
+    "balena-errors": "^4.7.1",
     "balena-hup-action-utils": "~4.0.2",
     "balena-pine": "^12.4.0",
     "balena-register-device": "^7.1.0",
-    "balena-request": "^11.1.1",
+    "balena-request": "^11.2.0-update-balena-auth-0622683ae2c02682f369b792a49e931fdc11468a",
```
Comment (pdcastro) [inline package.json:107]: This can now be updated to `"^11.2.0"`.

**Review by pdcastro — CHANGES_REQUESTED** [review]
Some Required tests are failing with the following errors:

**ResinCI/npm/store-npm — Npm package store failed**
```
[19:45:44] Starting 'pack-browser'...
[19:45:44] 'pack-browser' errored after 124 ms
[19:45:44] Error: 
/tmp/build/66d12d10/versioned-source/node_modules/balena-request/build/request.d.ts:5
import type BalenaAuth from 'balena-auth';
^
ParseError: 'import' and 'export' may appear only with 'sourceType: module'
    at formatError (/tmp/build/66d12d10/versioned-source/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^4.0.0/format-error.js:21:10)
    at Gulp.<anonymous> (/tmp/build/66d12d10/versioned-source/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^4.0.0/log/events.js:33:15)
    at Gulp.emit (events.js:327:22)
    at Gulp.EventEmitter.emit (domain.js:482:12)
    at Object.error (/tmp/build/66d12d10/versioned-source/node_modules/undertaker/lib/helpers/createExtensions.js:61:10)
    at handler (/tmp/build/66d12d10/versioned-source/node_modules/now-and-later/lib/map.js:50:14)
    at f (/tmp/build/66d12d10/versioned-source/node_modules/once/once.js:25:25)
    at f (/tmp/build/66d12d10/versioned-source/node_modules/once/once.js:25:25)
    at tryCatch (/tmp/build/66d12d10/versioned-source/node_modules/async-done/index.js:24:15)
    at done (/tmp/build/66d12d10/versioned-source/node_modules/async-done/index.js:40:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 build-es2015: `npm run build-es2015:fast && gulp pack-browser --buildDir es2015`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 build-es2015 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-11-11T19_45_44_966Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 build: `npm run lint && npm run build-es2015 && npm run build-es2018 && npm run docs`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-11-11T19_45_45_029Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 prepack: `npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the balena-sdk@15.18.0-update-balena-auth-8c9911f599e5e5c781a81a759bbee25414bff417 prepack script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-11-11T19_45_45_042Z-debug.log
...
```

**continuous-integration/appveyor/pr — AppVeyor build failed**
```
[19:41:41] Starting 'pack-browser'...
[19:41:42] 'pack-browser' errored after 191 ms
[19:41:42] Error: 
C:\projects\balena-sdk\node_modules\balena-request\build\request.d.ts:5
import type BalenaAuth from 'balena-auth';
^
ParseError: 'import' and 'export' may appear only with 'sourceType: module'
    at formatError (C:\projects\balena-sdk\node_modules\gulp\node_modules\gulp-cli\lib\versioned\^4.0.0\format-error.js:21:10)
    at Gulp.<anonymous> (C:\projects\balena-sdk\node_modules\gulp\node_modules\gulp-cli\lib\versioned\^4.0.0\log\events.js:33:15)
    at Gulp.emit (events.js:203:15)
    at Gulp.EventEmitter.emit (domain.js:448:20)
    at Object.error (C:\projects\balena-sdk\node_modules\undertaker\lib\helpers\createExtensions.js:61:10)
    at handler (C:\projects\balena-sdk\node_modules\now-and-later\lib\map.js:50:14)
    at f (C:\projects\balena-sdk\node_modules\once\once.js:25:25)
    at f (C:\projects\balena-sdk\node_modules\once\once.js:25:25)
    at tryCatch (C:\projects\balena-sdk\node_modules\async-done\index.js:24:15)
    at done (C:\projects\balena-sdk\node_modules\async-done\index.js:40:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! balena-sdk@15.17.0 build-es2015: `npm run build-es2015:fast && gulp pack-browser --buildDir es2015`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the balena-sdk@15.17.0 build-es2015 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2020-11-11T19_41_42_078Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! balena-sdk@15.17.0 build: `npm run lint && npm run build-es2015 && npm run build-es2018 && npm run docs`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the balena-sdk@15.17.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2020-11-11T19_41_42_110Z-debug.log
npm ERR! Test failed.  See above for more details.
Command exited with code 1
```

**Comment by pdcastro** [comment]
Hmm now we are back to the `Missing environment credentials` error:
```
Running SDK tests against: https://api.balena-cloud.com
TEST_USERNAME: undefined

/tmp/build/ca2460f8/versioned-source/tests/integration/setup.ts:82
		throw new Error('Missing environment credentials');
        ^
Error: Missing environment credentials
```

This error was previously attributed to the source branch being a fork. @thgreasi, do you know the reason for preventing `balena-sdk` builds from forks? Is it security, related to the test credentials? I remember that PRs for `balena-cli` also didn't work with forks in the past, but I think @nazrhom got it sorted. Would it be a case of "getting it sorted" for `balena-sdk` as well?

Otherwise, it sounds like we would have to create an identical PR on José's behalf, which does sound like an efficient and rewarding process...

**Comment by thgreasi** [comment]
Just pushed this commit in a branch to get the tests started.
See: https://github.com/balena-io/balena-sdk/pull/1019

Tests started at
See: https://ci.appveyor.com/project/resin-io/balena-sdk/builds/36355568

**Review by thgreasi — APPROVED** [review]

**Comment by thgreasi** [comment]
@pdcastro tests do pass. I think this is only blocked on your request-for-changes review.

**Review by pdcastro — APPROVED** [review]

**Comment by pdcastro** [comment]
@thgreasi, @nazrhom, it looks like versionbot merged this PR but never got to update the Changelog nor the npm registry - they should have been updated for v15.18.0, but they're still at v15.17.0. Would it again be because the source branch is a fork? Why is it that forks are not allowed?
The workaround, I guess, would be for us to merge another PR - e.g. #1019 which had been created just to get around the fork limitation, in relation to running tests.

**Comment by thgreasi** [comment]
@balena-ci republish

**Comment by thgreasi** [comment]
Thanks for the heads-up @pdcastro . It looks like we will need to wait for Giovanni to check this.

**Pull request merged**